src/fastgltf_tools.hpp

```diff
@@ -0,0 +1,303 @@
+		// have its target or byteStride properties defined."
+		auto valueStride = getElementByteSize(accessor.type, accessor.componentType);
+
+		auto idx = internal::getAccessorElementAt<std::uint32_t>(accessor.sparse->indexComponentType,
+				indicesBytes + indexStride * index);
+
+		return internal::getAccessorElementAt<ElementType>(accessor.componentType,
+				valuesBytes + valueStride * idx);
+	}
+	else {
+		auto& view = asset.bufferViews[*accessor.bufferViewIndex];
+		auto stride = view.byteStride ? *view.byteStride : getElementByteSize(accessor.type, accessor.componentType);
+
+		auto* bytes = adapter(asset.buffers[view.bufferIndex]);
+		bytes += view.byteOffset + accessor.byteOffset;
+
+		return internal::getAccessorElementAt<ElementType>(accessor.componentType, bytes + index * stride);
+	}
+}
+
+template <typename ElementType, typename Functor, typename BufferDataAdapter = DefaultBufferDataAdapter>
+void iterateAccessor(const Asset& asset, const Accessor& accessor, Functor&& func,
+		const BufferDataAdapter& adapter = {}) {
+	if (accessor.type != ElementTraits<ElementType>::type) {
+		return;
+	}
+
+	if (accessor.sparse) {
+		auto& indicesView = asset.bufferViews[accessor.sparse->bufferViewIndices];
+		auto* indicesBytes = adapter(asset.buffers[indicesView.bufferIndex])
+				+ indicesView.byteOffset + accessor.sparse->byteOffsetIndices;
+		auto indexStride = getElementByteSize(AccessorType::Scalar, accessor.sparse->indexComponentType);
+
+		auto& valuesView = asset.bufferViews[accessor.sparse->bufferViewValues];
+		auto* valuesBytes = adapter(asset.buffers[valuesView.bufferIndex])
+				+ valuesView.byteOffset + accessor.sparse->byteOffsetValues;
+		// "The index of the bufferView with sparse values. The referenced buffer view MUST NOT
+		// have its target or byteStride properties defined."
+		auto valueStride = getElementByteSize(accessor.type, accessor.componentType);
+
+		for (std::size_t i = 0; i < accessor.sparse->count; ++i) {
+			auto idx = internal::getAccessorElementAt<std::uint32_t>(accessor.sparse->indexComponentType,
+					indicesBytes + indexStride * i);
+			func(internal::getAccessorElementAt<ElementType>(accessor.componentType,
+					valuesBytes + valueStride * idx));
+		}
+	}
+	else {
+		auto& view = asset.bufferViews[*accessor.bufferViewIndex];
+		auto stride = view.byteStride ? *view.byteStride : getElementByteSize(accessor.type, accessor.componentType);
+
+		auto* bytes = adapter(asset.buffers[view.bufferIndex]);
+		bytes += view.byteOffset + accessor.byteOffset;
+
+		for (std::size_t i = 0; i < accessor.count; ++i) {
+			func(internal::getAccessorElementAt<ElementType>(accessor.componentType, bytes + i * stride));
+		}
+	}
+
+}
+
+template <typename ElementType, std::size_t TargetStride = sizeof(ElementType),
+		 typename BufferDataAdapter = DefaultBufferDataAdapter>
+void fillTargetFromAccessor(const Asset& asset, const Accessor& accessor, void* dest,
```

Comment:
I feel like `copyFromAccessor` is a much better general name for this function.

Reply:
Resolved in [bc67bc4](https://github.com/spnda/fastgltf/pull/20/commits/bc67bc47bf712fc37adb1fbdd9f92ddeb943cab4)